modules/quantcastIdSystem.js

```diff
@@ -7,11 +7,153 @@
 
 import {submodule} from '../src/hook.js'
 import { getStorageManager } from '../src/storageManager.js';
+import { triggerPixel, logInfo } from '../src/utils.js';
+import { uspDataHandler, coppaDataHandler, gdprDataHandler } from '../src/adapterManager.js';
 
 const QUANTCAST_FPA = '__qca';
+const DEFAULT_COOKIE_EXP_TIME = 392; // (13 months - 2 days)
+const PREBID_PCODE = 'p-KceJUEvXN48CE'; // Not associated with a real account
+const DOMAIN_QSERVE = 'https://pixel.quantserve.com/pixel';
+const QUANTCAST_VENDOR_ID = '11';
+const PURPOSE_DATA_COLLECT = '1';
+const PURPOSE_PRODUCT_IMPROVEMENT = '10';
+const QC_TCF_REQUIRED_PURPOSES = [PURPOSE_DATA_COLLECT, PURPOSE_PRODUCT_IMPROVEMENT];
+const QC_TCF_CONSENT_FIRST_PURPOSES = [PURPOSE_DATA_COLLECT];
+const QC_TCF_CONSENT_ONLY_PUPROSES = [PURPOSE_DATA_COLLECT];
+const GDPR_PRIVACY_STRING = gdprDataHandler.getConsentData();
+const US_PRIVACY_STRING = uspDataHandler.getConsentData();
 
 export const storage = getStorageManager();
 
+export function firePixel(clientId, cookieExpTime = DEFAULT_COOKIE_EXP_TIME) {
+  // check for presence of Quantcast Measure tag _qevent obj,
+  // required gdpr consent / legitimate interest for purpose 1 and 10
+  // for Quantcast as a vendor, and publisher provided clientID
+  if (!window._qevents && hasGDPRConsent(GDPR_PRIVACY_STRING) && clientId && clientId != '') {
+    var fpa = storage.getCookie(QUANTCAST_FPA);
+    var fpan = '0';
+    var domain = quantcastIdSubmodule.findRootDomain();
+    var now = new Date();
+    var usPrivacyParamString = '';
+    var firstPartyParamStrings;
+    var gdprParamStrings;
+
+    if (!fpa) {
+      var expires = new Date(now.getTime() + (cookieExpTime * 86400000)).toGMTString();
+      fpa = 'B0-' + Math.round(Math.random() * 2147483647) + '-' + et;
+      fpan = '1';
+      storage.setCookie(QUANTCAST_FPA, fpa, expires, '/', domain, null);
+    }
+
+    firstPartyParamStrings = `&fpan=${fpan}&fpa=${fpa}`;
+    gdprParamStrings = '&gdpr=0';
+    if (GDPR_PRIVACY_STRING && typeof GDPR_PRIVACY_STRING.gdprApplies === 'boolean' && GDPR_PRIVACY_STRING.gdprApplies) {
+      gdprParamStrings = `gdpr=1&gdpr_consent=${GDPR_PRIVACY_STRING.consentString}`;
+    }
+    if (US_PRIVACY_STRING && typeof US_PRIVACY_STRING === 'string') {
+      usPrivacyParamString = `&us_privacy=${US_PRIVACY_STRING}`;
+    }
+
+    let url = DOMAIN_QSERVE +
```

Comment:
Shouldn't we also capture the local client time and timezone? (And screen resolution, if legal agrees).

Reply:
I just had removed 'et' and 'tzo' in the previous commit - https://github.com/sachinsurfs/Prebid.js/pull/3/commits/b31c0f9e75f432dd16f938ebe7a994be4f540d52

Only cause the general consensus currently is that we want to keep the number of parameters we send through prebid minimum in our first iteration unless we have strong justification for needing it. We currently are not using local client time / entry time and timezone offset in Pixelserver / HS / Adrain and modeling team are not using these values in any production systems and instead relying on `x_timestamp`, so I thought we can add it later if needed.